telegram/ext/_picklepersistence.py

```diff
@@ -49,47 +47,34 @@ def _reconstruct_to(cls: Type[TelegramObj], kwargs: dict) -> TelegramObj:
     """
     This method is used for unpickling. The data, which is in the form a dictionary, is
     converted back into a class. Works mostly the same as :meth:`TelegramObject.__setstate__`.
-    This function should be kept in place for backwards compatibility even if the pickling logic
-    is changed, since `_custom_reduction` places references to this function into the pickled data.
```

Comment:
Why was this comment removed? The contents `_custom_recdution` of just went into `reducer_override` and this still inserts a refercne to `_reconstruct_to` …

Reply:
I'm confused by the statement "This function should be kept in place for backwards compatibility". This function needs to be there for 3.8+ as well, since the return of `reducer_override` [needs](https://docs.python.org/3/library/pickle.html#object.__reduce__) a callable object anyway.

Reply:
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/22366557/d5796868-aef0-45d8-acb8-388d0e8839ec)

You're confused? ;P Regardless of other reasons to keep this method, the point is the following:

1. you pickle stuff. while pickling, a reference to `telegram.ext._picklepersisence.__reconstruct_to` is inserted
2. We upgrade you the library, removing `__reconstruct_to`
3. you update ptb locally and try to load the data
4. reference to `telegram.ext._picklepersisence.__reconstruct_to` can not be resolved
5. 💥

Reply:
> ![image](https://user-images.githubusercontent.com/22366557/243198762-d5796868-aef0-45d8-acb8-388d0e8839ec.png)
> 
> You're confused? ;P

weren't my words 🫣 https://github.com/python-telegram-bot/python-telegram-bot/pull/2893#discussion_r808418514, 
I had thought that comment pertained to the py3.7 special case

> the point is the following:

Ah I get it now, I just thought that python internally did something different but now that I say that out loud I realize that sounds stupid haha

I'll add back the _custom_reduction function again.

Reply:
> weren't my words

nononononono, only git blame matters ;P

👍